contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,583 @@
+    }
+
+    /**
+     * Redeems a set token and sells the underlying tokens using Uniswap
+     * or Sushiswap.
+     *
+     * @param _setToken             Address of the set token being redeemed
+     * @param _amountSetToRedeem    The amount of the set token to redeem
+     * @param _isOutputETH          Set to true if the output token is Ether
+     * @param _outputToken          Address of output token. Ignored if _isOutputETH is true
+     * @param _minOutputReceive     Minimum amount of output token / ether to receive
+     */
+    function exchangeRedeem(
+        ISetToken _setToken,
+        uint256 _amountSetToRedeem,
+        bool _isOutputETH,
+        address _outputToken,
+        uint256 _minOutputReceive
+    ) 
+        external
+        nonReentrant
+    {
+        _setToken.transferFrom(msg.sender, address(this), _amountSetToRedeem);
+        basicIssuanceModule.redeem(_setToken, _amountSetToRedeem, address(this));
+        _liquidateComponents(_setToken);
+        uint256 outputAmount = _handleRedeemOutput(_isOutputETH, _outputToken, _minOutputReceive);
+        emit ExchangeRedeem(msg.sender, address(_setToken), _isOutputETH ? address(0) : _outputToken, _amountSetToRedeem, outputAmount);
+    }
+
+    receive() external payable {}
+
+    /**
+     * Returns an estimated quantity of the specified SetToken given an input amount of ETH or a specified ERC20 receieved when issuing.
+     * Estimating pulls the best price of each component using Uniswap or Sushiswap
+     *
+     * @param _setToken         Address of the set token being issued
+     * @param _amountInput      Amount of the input token to spend
+     * @param _isInputETH       Set to true if the input token is Ether
+     * @param _inputToken       Address of input token. Ignored if _isInputETH is true
+     * @return                  Estimated amount of Set tokens that will be received
+     */
+    function getEstimatedIssueSetQuantity(
+        ISetToken _setToken,
+        uint256 _amountInput,
+        bool _isInputETH,
+        IERC20 _inputToken
+    )
+        external
+        view
+        returns (uint256)
+    {
+        uint256 amountEth;
+        if(!_isInputETH && address(_inputToken) != WETH) {
+            uint256 uniAmount = _tokenAvailable(uniFactory, address(_inputToken)) ? _getSellPrice(true, address(_inputToken), _amountInput) : 0;
+            uint256 sushiAmount = _tokenAvailable(sushiFactory, address(_inputToken)) ? _getSellPrice(false, address(_inputToken), _amountInput) : 0; 
+            amountEth = Math.max(uniAmount, sushiAmount);
+        } else {
+            amountEth = _amountInput;
+        }
+
+        // get price of set token on uniswap
+        uint256 wethBalance = amountEth;
+        (uint256 tokenReserveA, uint256 tokenReserveB) = UniswapV2Library.getReserves(uniFactory, WETH, address(_setToken));
+        uint256 minSetTokenAmountOut = uniRouter.getAmountOut(wethBalance, tokenReserveA, tokenReserveB);
```

Comment:
Same here, we can include `sushiwap.getAmountOut()` here and set `minSetTokenAmountOut` as `max(uniTokenAmountOut, sushiTokenAmountOut`.